functions/src/export-geojson.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright 2025 The Ground Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as functions from 'firebase-functions';
+import {Map} from 'immutable';
+import JSONStream from 'jsonstream-ts';
+import {canExport, canImport} from './common/auth';
+import {getDatastore} from './common/context';
+import * as HttpStatus from 'http-status-codes';
+import {DecodedIdToken} from 'firebase-admin/auth';
+import {toMessage} from '@ground/lib';
+import {GroundProtos} from '@ground/proto';
+import {toGeoJsonGeometry} from '@ground/lib';
+
+import Pb = GroundProtos.ground.v1beta1;
+
+/**
+ * Iterates over all LOIsin a job returning a valid GeoJSON file.
+ */
+export async function exportGeojsonHandler(
```

Comment:
This is quite a long function. This is a nit, but any way this could be split up into useful chunks of logic? e.g. error checking, writing rows, etc.